File: src/Core/Repository.cs
using System;
using System.Collections.Generic;

namespace NPackage.Core
{
    public class Repository
    {
        private readonly List<Uri> repositoryImports = new List<Uri>();
        private readonly List<Uri> packageImports = new List<Uri>();
        private readonly List<Package> packages = new List<Package>();

        public IList<Uri> PackageImports
        {
            get { return packageImports; }
        }

        public IList<Uri> RepositoryImports
        {
            get { return repositoryImports; }
        }

        public IList<Package> Packages
        {
            get { return packages; }
        }
    }
}

File: src/Core/Repository.fs
namespace NPackage.Core

open System
open System.Collections.Generic

type Repository() =
    let repositoryImports = new ResizeArray<string>()
    let packageImports = new ResizeArray<string>()
    let packages = new ResizeArray<Package>()

    member this.PackageImports = packageImports :> IList<string>
    member this.RepositoryImports = repositoryImports :> IList<string>
    member this.Packages = packages :> IList<Package>
